KthSmallestElement/Solution.cpp

```diff
@@ -0,0 +1,18 @@
+//
+// Created by anetreba on 07.07.21.
+//
+
+#include "Solution.h"
+
+int Solution::kthSmallest(std::vector <std::vector<int>> &matrix, int k) {
+    if (matrix.empty() || k > matrix.size() * matrix[0].size() || k < 1)
+        return 0;
+    std::vector<int> arr;
+
+    for (auto &i : matrix)
+        for (auto &j : i)
+            arr.push_back(j);
+    std::sort(arr.begin(), arr.end());
+    return arr[k - 1];
+}
+
```

Comment:
this alghoritm works fine but you haven't use info that each row and column already sorted in ascending order. Think about binary search. It should speed up complexity from linear to log